ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -663,6 +592,9 @@ void addReplicasToPartition(AmbryPartition partition, List<AmbryReplica> replica
         return v;
       });
       clusterWideAllocatedRawCapacityBytes.getAndAdd(replicas.get(0).getCapacityInBytes() * replicas.size());
+      for (ClusterMapChangeListener listener : clusterMapChangeListeners) {
+        listener.onReplicaAddedOrRemoved(replicas, Collections.emptyList());
```

Comment:
After rebasing master branch, we should be able to remove this. In #1405, clustermap change listeners are called within DynamicClusterChangeHandler. If you have additional concern, we can talk offline.

Reply:
One concern now is that we can't add a listener from the background startup thread since the DcInfo object for other datacenters may not be present yet. Here I can either add the listener after all of the startup threads are joined, or make a new type of listener that is notified on new partitions added.

Reply:
I chose to connect the listeners after the other startup threads are finished.